app/src/main/java/fi/aalto/legroup/achso/entities/OptimizedVideo.java

```diff
@@ -0,0 +1,359 @@
+package fi.aalto.legroup.achso.entities;
+
+import android.graphics.PointF;
+import android.location.Location;
+import android.net.Uri;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import fi.aalto.legroup.achso.storage.VideoRepository;
+
+/**
+ * Store videos in a more GC friendly way. Should be equivalent with the Video class.
+ * You should use this class directly as much as possible and inflating to Video only when necessary.
+ *
+ * These should round trip:
+ *   new OptimizedVideo(video).inflate()
+ *   new OptimizedVideo(optimizedVideo.inflate())
+ */
+public class OptimizedVideo {
+
+    private transient String manifestUri;
+    private transient VideoRepository repository;
```

Comment:
I guess the User pooling makes them _very bad (tm)_ for serialization, maybe I should remove them.